doc/arch/adr-005-api-client-app-auth.md

```diff
@@ -0,0 +1,121 @@
+RWD endpoints: the [Geoprocessing API.](https://github.com/WikiWatershed/model-my-watershed/blob/develop/doc/arch/adr-004-geoprocessing-api.md) The analyze and RWD endpoints have
+have allow-any access; we'd now like to secure them via token authentication
+so that we can better track individual users, prevent users from sending off too many
+requests, revoke the access of problematic users, and some day maybe add a 
+paid tier of use.
+
+Enforcing token authentication on these previously open endpoints causes
+a problem for our client app; the client app consumes the same API, and
+we want to allow it to use these endpoints un-authenticated. We need a way 
+to identify the client app as a special case from which the app should allow 
+unrestricted requests requests.
+
+Because everything we send to and from the client is exposed to the app user,
+any mechanism we give the client app to identify itself will be fairly
+easy to uncover and then fake by anyone trying to get un-checked access to the
+API. With this in mind our solution should prioritize not be overly complicated/difficult
+for us when we know there's no truly secure solution (barring doing all our rendering
+server-side).
+
+Some possibilities are:
+
+1. Give each environment of the client app its own API token
+   - Keeps a single system for API authentication; easy to reason about;
+     as a bonus will allow us to cycle the client app's token if someone
+     decides to start using it
+   
+1. Set a special, is-client-app flag that the API would check to determine if authentication was necessary
+   - To set the flag we'd have to know the request was from the client app in
+     the first place. Preliminary checks for doing this via `http_referer`
+     were unsuccessful, which would leave doing this via custom header (or some
+     other part of the request). Setting a `X-IsClientApp` custom header might
+     be overly naive or hacky, and could result in a lot of conditional logic
+
+
+## Decision
+
+**Give the client app its own API token**
+
+Setting up the client app with its own API token will result a cleaner
+architecture for the API. All users use a token, no exceptions. In addition to
+the simpler mental model the API token method provides, we'll also gain an added
+level of token enforcement; if some API user tries to programmatically use the client
+app's token instead of their own, we'll be able to easily cycle the client app's token 
+to de-incentivize them. 
+
+One easy way to cycle the token:
+```
+./manage.py drf_create_token -r <client_app_username>
+```
+
+##### When should the client app send its token
+
+For all Geoprocessing API requests, whether there's a logged-in user's token available
+to use or not, ie., we will continue to use a user's credentials only for user/model/project
```

Comment:
Maybe this is just being pedantic, but technically those requests don't need any credentials to access.  Maybe refer to these as application endpoints, to distinguish them from "api" endpoints?